controlplane/kubeadm/internal/webhooks/kubeadm_control_plane.go

```diff
@@ -455,6 +462,37 @@ func validateRolloutStrategy(rolloutStrategy *controlplanev1.RolloutStrategy, re
 	return allErrs
 }
 
+func validateNamingStrategy(machineNamingStrategy *controlplanev1.MachineNamingStrategy, pathPrefix *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if machineNamingStrategy.Template != "" {
+		if !strings.Contains(machineNamingStrategy.Template, ".random") {
+			allErrs = append(allErrs,
+				field.Invalid(
+					pathPrefix.Child("template"),
+					machineNamingStrategy.Template,
+					"invalid ControlPlane name template, {{ .random }} variable is missed",
+				))
+			return allErrs
+		}
+		name, err := names.KCPMachineNameGenerator(machineNamingStrategy.Template, "cluster", "kubeadmcontrolplane").GenerateName()
+		if err != nil {
+			allErrs = append(allErrs,
+				field.Invalid(
+					pathPrefix.Child("template"),
+					machineNamingStrategy.Template,
+					fmt.Sprintf("invalid ControlPlane name template: %v", err),
+				))
+		} else {
+			for _, err := range validation.IsDNS1123Subdomain(name) {
+				allErrs = append(allErrs, field.Invalid(pathPrefix.Child("template"), machineNamingStrategy.Template, err))
```

Comment:
```suggestion
				allErrs = append(allErrs,
					field.Invalid(
						pathPrefix.Child("template"),
						machineNamingStrategy.Template,
						fmt.Sprintf("invalid template, generated names would not be valid Kubernetes object names: %v", err),
					))
```
Let's add some context